File: test/fixtures/constevents/ios/Sources/ConstEventsPlugin.swift
import Foundation
import Capacitor

@objc(ConstEventsPlugin)
public class ConstEventsPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "ConstEventsPlugin"
    public let jsName = "ConstEvents"
    public let pluginMethods: [CAPPluginMethod] = []

    private let stateChangeEvent = "stateChange"
    private let urlOpenEvent = "urlOpen"

    @objc func exit(_ call: CAPPluginCall) {
        notifyListeners(stateChangeEvent, data: ["active": false], retainUntilConsumed: true)
        notifyListeners(urlOpenEvent, data: ["url": "https://example.com"], retainUntilConsumed: true)
        call.resolve()
    }
}

File: test/fixtures/constevents/android/src/main/java/com/example/constevents/ConstEventsPlugin.java
package com.example.constevents;

import com.getcapacitor.JSObject;
import com.getcapacitor.PermissionCallback;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "ConstEvents")
public class ConstEventsPlugin extends Plugin {

    private static final String EVENT_STATE_CHANGE = "stateChange";
    private static final String EVENT_PAUSE = "pause";

    @PluginMethod
    public void exit(PluginCall call) {
        JSObject data = new JSObject();
        data.put("active", false);
        notifyListeners(EVENT_STATE_CHANGE, data, true);
        notifyListeners(EVENT_PAUSE, null, false);
        call.resolve();
    }

    @PermissionCallback
    private void permsCallback(PluginCall call) {
        call.resolve();
    }
}
